Release RC2 (#271)
JoshOrndorff/recipes: text/3-entrees/storage-api/cache.md

# Cache Multiple Calls

_[`pallets/storage-cache`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/storage-cache)_

> Calls to runtime storage have an associated cost. With this in mind, multiple calls to storage
> values should be avoided when possible.

Reviewer: ```suggestion
Calls to runtime storage have an associated cost and developers should strive to minimize the number of calls.
```

## Copy Types

For [`Copy`](https://doc.rust-lang.org/std/marker/trait.Copy.html) types, it is easy to reuse

> previous storage calls by simply reusing the value (which is automatically cloned upon reuse). With
> this in mind, the second call in the following code is unnecessary:

Reviewer: ```suggestion
previous storage calls by simply reusing the value (which is automatically cloned upon reuse). In the code below, the second call is unnecessary:
```

## Clone Types

If the type was not `Copy`, but was [`Clone`](https://doc.rust-lang.org/std/clone/trait.Clone.html),

> then it is still preferred to clone the value in the method than to make another call to runtime

Reviewer: ```suggestion
then it is still better to clone the value in the method than to make another call to runtime
```
